test_framework/script.py

```diff
@@ -1097,9 +1097,10 @@ def generate_threshold_csa(k, pubkeys):
         return tapscripts
 
 class TapTree:
-    def __init__(self):
-        self.root = Node()
-        self.key = ECPubKey()
+    def __init__(self, *, root=None, key=None):
+        """Taptree constructor. Takes an optional `root` Tapbranch and `key` ECPubKey."""
```

Comment:
nit: Could also be a tapscript object as in chapter 2.3.

Reply:
nit: Swapping order of args to "key, root " would be consistent with taptree descriptor order (tp(key, tree))

Reply:
thanks. Good suggestions.

I'll swap the order of the args so the definition here matches the descriptor format. Note however that the special `*` argument means that any arguments after need to be passed in as named args, so order doesn't matter for the callers.